bundles/admin/admin-permissions/handler/layerRightsHandler.js

```diff
@@ -39,6 +45,57 @@ class UIHandler extends StateHandler {
         });
     }
 
+    setCheckAllForPermission (permissionType, enabled) {
+        let permissions = [...this.state.resources];
+        const startIndex = (this.state.pagination.page - 1) * this.state.pagination.pageSize;
+        const endIndex = this.state.pagination.pageSize * this.state.pagination.page;
+        const changedIds = new Set(this.state.changedIds);
+        for (let i = startIndex; i < endIndex && i < permissions.length; i++) {
+            const permIndex = permissions[i].permissions.findIndex(p => p.id === permissionType);
+            permissions[i].permissions[permIndex].allow = enabled;
+            changedIds.add(permissions[i].id);
+        }
+
+        this.updateState({
+            resources: permissions,
+            changedIds: new Set(changedIds)
+        });
+    }
+
+    setPage (page) {
+        this.updateState({
+            pagination: {
+                ...this.state.pagination,
+                page: page
+            }
+        });
+    }
+
+    search (searchText) {
+        const permissions = structuredClone(this.state.permissions?.resource?.filter(r => r.name.toLowerCase().includes(searchText.toLowerCase())));
+        this.updateState({
+            resources: permissions,
+            changedIds: new Set(),
```

Comment:
Why does searching and clearing search change the changedIds? Aren't those the layers that have permissions changed after saving or what is tracked with it?

Reply:
That tracks which layers have been edited and need to be saved, and search clears unsaved changes.